### PR TITLE
fix: Fix serde deserialization for FinalExecutionOutcomeViewEnum

### PR DESCRIPTION
**Pull request by kiseln**
Discussion: https://near.zulipchat.com/#narrow/stream/297873-node/topic/An.20issue.20with.20FinalExecutionOutcomeView

We had to switch the order of the variants because `FinalExecutionOutcome` is a subset of `FinalExecutionOutcomeWithReceipt` which means serde could never resolve to the latter.

Borsh attributes were dropped as per discussion. While technically a breaking change, it seems like this enum is only intended to be used with JSON-RPC endpoint. Therefore, we think it's unlikely that someone serialized it with borsh.

### CODE REVIEW

**Comment by frol** [comment]
@kiseln Thanks for the contribution! If CI passes, it is good to go from my point of view as per the view I shared in Zulip conversation.

@race-of-sloths invite

**Comment by race-of-sloths** [comment]
@frol Thank you for calling!
    
@kiseln Thank you for the contribution! Join [Race of Sloths](https://race-of-sloths.com) by simply mentioning me in your comment/PRs description and start collecting Sloth Points through contributions to open source projects.

<details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/kiseln)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: core/primitives/src/views.rs
Comment (akhi3030) [inline core/primitives/src/views.rs:1752]: Could you please a comment here mentioning that the ordering is relevant and why.

**Review by akhi3030 — APPROVED** [review]

**Comment by akhi3030** [comment]
@kiseln: ping me when you are ready to have this PR merged and I'll click the button for you.

File: core/primitives/src/views.rs
Comment (kiseln) [reply core/primitives/src/views.rs:1754]: I did, plus added tests that I think are useful

**Comment by kiseln** [comment]
@akhi3030 could you please approve again?

**Pull request merged**
